File: javascripts/power_recursion.js
var power = function (x,n){
    if(n === 0){ //zero
        return 1;
    }
    else if(n < 0){ //reciprocal
        var a = power(x, -n);
        return 1/a;
    }
    else if((n%2) === 0){ //even
        var y = power(x, n/2);
        return y*y;
    }
    else if((n%2) !== 0){ //odd
        return x*power(x, n-1);
    }  
};
console.log(power(3,-1));

File: python scripts/power_recursion.py
def power(x,n):
    if(n == 0):
        return 1
    if(n < 0):
        return 1/power(x, -n)
    if(n%2 == 0):
        y = power(x, n/2)
        return y*y
    if(n%2 != 0):
        return x*power(x, n-1)

print(power(25,-1))